packages/lesswrong/components/Layout.tsx

```diff
@@ -323,7 +333,10 @@ class Layout extends PureComponent<LayoutProps,LayoutState> {
 const LayoutComponent = registerComponent<ExternalProps>(
   'Layout', Layout, { styles, hocs: [
     withLocation, withCookies,
-    withUpdateCurrentUser,
+    withUpdate({
+      collectionName: "Users",
+      fragmentName: 'UsersCurrent',
+    }),
```

Comment:
Curious to know why we needed to switch away from updateCurrentUser?

Reply:
Turns out the existing usage of `withUpdateCurrentUser` was actually broken, and the second usage added failed testing immediately, because it calls `useCurrentUser` which relies on a context provider that `Layout` provides to its children, which isn't available inside `Layout` itself. `withUpdate` works around this problem by taking the user as a parameter, rather than trying to get it from a context provider.

Reply:
Makes sense.